app/controllers/auth_controller.py

```diff
@@ -1,45 +1,47 @@
-from typing import Tuple
+from typing import Any
 
-from flask import request, Blueprint, render_template, redirect
+from flask import request, Blueprint, render_template, redirect, url_for
 
 from app.forms.signup_form import SignupForm
 from app.services import auth_service
-from app.utils.session_decorators import login_required
+from app.utils.session_decorators import login_required, not_logged_in_required
 from app.utils.validation import is_email_valid
 
 auth_routes = Blueprint('auth_routes', __name__)
 
 
 @auth_routes.route('/signup', methods=['POST', 'GET'])
-def sign_up() -> Tuple[str, int]:
-    form = SignupForm();
+@not_logged_in_required
+def sign_up() -> Any:
+    form = SignupForm()
 
     if form.validate_on_submit():
```

Comment:
Email validation should check if email is already signed up.

Reply:
Yea so that is done in the form class if you take a look at `forms/sugnup_form.py` near the bottom you will see a `validate_email` func

Reply:
All good then